### PR TITLE
chore: add domain file translation validation

### LINKED ISSUE #112: Validate translations in domain files

### PR DESCRIPTION
**Pull request by dmorand**
## Description

Add translation check when performing domain file validation

## Related issues

* Closes #112

## Checklist

- [ ] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) guidelines 
- [ ] All relevant PR sections are populated, irrelevant ones are removed

### CODE REVIEW

File: core/scripts/validate_domain.py [hunk 1]
```diff
@@ -9,48 +9,59 @@
...
+    missing_french_response_ids = english_response_ids - french_response_ids
 
-        if missing_english_response_ids:
-            print(f"Missing english responses: {missing_english_response_ids}")
+    if missing_english_response_ids:
+        print(f"Missing english responses: {missing_english_response_ids}")
 
-        if missing_english_response_ids:
-            print(f"Missing french responses: {missing_french_response_ids}")
+    if missing_english_response_ids:
```
Comment (NetBKarine) [inline core/scripts/validate_domain.py:55]: Copy-paste much?

File: core/scripts/validate_domain.py [hunk 2]
```diff
@@ -9,48 +9,59 @@
...
 BUTTONS_KEY = "buttons"
+TEXT_KEY = "text"
+TITLE_KEY = "title"
 PAYLOAD_KEY = "payload"
 
 
-def get_buttons(response: List[dict]) -> List[str]:
+def get_text(response: List[dict]) -> Set[str]:
     if len(response) != 1:
         raise Exception("This tool does not support checking response variants (yet).")
```
Comment (NetBKarine) [inline core/scripts/validate_domain.py:19]: Shouldn't that only be a warning?

File: core/scripts/validate_domain.py [hunk 3]
```diff
@@ -61,8 +72,48 @@ def validate_domain() -> int:
...
+        if response_id not in french_responses:
+            continue
+
+        english_response = english_responses[response_id]
+        french_response = french_responses[response_id]
+
+        english_text = get_text(english_response)
+        french_text = get_text(french_response)
+
+        if len(english_text - french_text) != len(english_text):
```
Comment (NetBKarine) [inline core/scripts/validate_domain.py:93]: Why not only `if english_text == french_text`? I trust there's a reason, but I don't see it.

File: core/scripts/validate_domain.py (hunk 2 above)
Comment (dmorand) [reply core/scripts/validate_domain.py:19]: A warning will fail silently in the CI, if we want to use variants, we should update the tool accordingly and be notified as soon as possible.

File: core/scripts/validate_domain.py (hunk 1 above)
Comment (dmorand) [reply core/scripts/validate_domain.py:55]: Good catch!

File: core/scripts/validate_domain.py (hunk 3 above)
Comment (dmorand) [reply core/scripts/validate_domain.py:93]: We are comparing a list of text elements (including the buttons) and we want to output a warning as soon as there is one text element not translated for the response. I will rename the variables (and the method name) to make this more obvious.

**Pull request merged**